### PR TITLE
task1

### PR DESCRIPTION
**Pull request by noakrams**
checked extreme cases with numbers represented at 32 bits, seems to be working fine.

Signed-off-by: noakrams <noakrams96@gmail.com>

### CODE REVIEW

File: src/asm_task1Assignment1.s [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        jmp printZ                          ; jump to lable printZ and skip the subtraction part
+        xIsBigger:
+                sub ecx, ebx                ; subtract arguments in case ecx is bigger than ebx
+        printZ:
+                pushad                      ; save registers before calling c function printf
+                pushfd
+                push ecx                    ; push ecx to the stack so we can print it
+                push dword format           ; push the format before printf
+                call printf                 ; print the result by printf c function
+                add esp, 8                  ; free space allocated for function arguments in stack
```
Comment (IdanTo1) [inline src/asm_task1Assignment1.s:40]: didn't understand this line

File: src/asm_task1Assignment1.s [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                pop ecx                     ; restore ecx
```
Comment (IdanTo1) [inline src/asm_task1Assignment1.s:41]: you can just pop because you don't use ecx

File: src/asm_task1Assignment1.s (hunk 1 above)
Comment (noakrams) [reply src/asm_task1Assignment1.s:40]: We need to remove format and ecx before the pop commands

File: src/main_task1Assignment1.c [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+int main(int argc, char **argv)
+{
+    int num1, num2;
+    char buf1[14];
+    char buf2[14];
+    printf("%s\n", "Please enter two integers");
+    fgets(buf1, 13, stdin);
+    fgets(buf2, 13, stdin);
+    sscanf(buf1, "%d", &num1);
+    sscanf(buf2, "%d", &num2);
```
[the remark this answers is not in the document]
Comment (noakrams) [reply src/main_task1Assignment1.c:16]: Not sure I understand what you mean

File: src/asm_task1Assignment1.s [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        pushad                              ; push all signficant registers onto stack (backup registers values) before c function
+        pushfd
+        push ebx                            ; push arguments in reverse order
+        push ecx                            ; push arguments in reverse order
+        call c_checkValidity                ; calling c function c_checkValidity
+        mov [isValid], eax                  ; move the result of c_checkValidity to the label isValid
+        pop ecx                             ; pulling the registers back
+        pop ebx                             ; pulling the registers back
+        popfd
+        popad                               ; restore all previously used registers
```
[the remark this answers is not in the document]
Comment (noakrams) [reply src/asm_task1Assignment1.s:27]: I wasn't sure about it so I asked Ben Hasson and he said it should be done that way

File: src/asm_task1Assignment1.s (hunk 1 above)
Comment (IdanTo1) [reply src/asm_task1Assignment1.s:40]: right! my bad!

File: src/main_task1Assignment1.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (IdanTo1) [reply src/main_task1Assignment1.c:17]: If i'm not wrong, the `fgets` method inserts to the buffer the newline character
so you need to `sscanf` for "%d\n"

File: src/asm_task1Assignment1.s [hunk 5]
```diff
@@ -16,30 +16,29 @@ assFunc:                                    ; assFun function definition
...
         jmp printZ                          ; jump to lable printZ and skip the subtraction part
         xIsBigger:
                 sub ecx, ebx                ; subtract arguments in case ecx is bigger than ebx
         printZ:
                 pushad                      ; save registers before calling c function printf
-                pushfd
+                pushfd                      ; backup EFLAGS
                 push ecx                    ; push ecx to the stack so we can print it
-                push dword format           ; push the format before printf
+                push format                 ; push the format before printf
```
Comment (IdanTo1) [inline src/asm_task1Assignment1.s]: when I think about it (and after asking Ben yesterday 😅 ) format is a pointer, and so it is 4-byte wide
so you were right all along (:
change it back to what is was (:

**Pull request merged**